gpflow/kernels/base.py

```diff
@@ -117,6 +121,42 @@ def slice_cov(self, cov: tf.Tensor) -> tf.Tensor:
 
         return cov
 
+    def _is_ard(self, raw_parameter):
+        is_ard = isinstance(raw_parameter, Iterable)
+        if (is_ard
+                and self.active_dims is not None
+                and not isinstance(self.active_dims, slice)
+                and len(raw_parameter) != len(self.active_dims)):
+            raise ValueError(f"Dimension of `active_dims` {self.active_dims} does not match "
+                             f"dimension of the ard parameter ({len(raw_parameter)})")
+        return is_ard
+
+    def validate_ard_parameter(self, X, parameter):
```

Comment:
This is wierd function, it just returns True? Why did you add this TODO and commented out code?

Reply:
I have removed this altogether for now as I cannot figure out how to perform this validation correctly during graph execution.